### PR TITLE
A few timed mines functions

### PR DESCRIPTION
**Pull request by d3nwah**
Boom!

### CODE REVIEW

File: Source/AliveLibAO/TimedMine.cpp [hunk 1]
```diff
@@ -184,6 +187,155 @@ __int16 TimedMine::VTakeDamage_408B90(BaseGameObject* pFrom)
...
+    field_1B8 |= 1;
+    if (sCollisions_DArray_504C6C->RayCast_40C410(
+        field_A8_xpos,
+        field_AC_ypos - FP_FromInteger(20),
+        field_A8_xpos,
+        field_AC_ypos + FP_FromInteger(20),
+        &pLine,
+        &hitX,
+        &hitY,
+        (field_BC_sprite_scale != FP_FromDouble(0.5)) ? 7 : 112))
```
Comment (paulsapps) [inline Source/AliveLibAO/TimedMine.cpp:241]: 112 would be better in hex

File: Source/AliveLibAO/TimedMine.cpp [hunk 2]
```diff
@@ -184,6 +187,155 @@ __int16 TimedMine::VTakeDamage_408B90(BaseGameObject* pFrom)
...
+                        PSX_RECT pObjRect = {};
+                        auto pLiftPoint = static_cast<LiftPoint*>(pObj);
+                        pLiftPoint->VGetBoundingRect_418120(&pObjRect, 1);
+                        if (FP_GetExponent(field_A8_xpos) > pObjRect.x &&
+                            FP_GetExponent(field_A8_xpos) < pObjRect.w &&
+                            FP_GetExponent(field_AC_ypos) < pObjRect.h)
+                        {
+                            field_F8_pLiftPoint = pLiftPoint;
+                            pLiftPoint->VAdd(this);
+                            ++field_F8_pLiftPoint->field_C_refCount;
```
Comment (paulsapps) [inline Source/AliveLibAO/TimedMine.cpp:266]: all the others of these are done post increment

File: Source/AliveLibAO/TimedMine.cpp [hunk 3]
```diff
@@ -184,6 +187,155 @@ __int16 TimedMine::VTakeDamage_408B90(BaseGameObject* pFrom)
...
+        {
+            field_1B0 = gnFrameCount_507670;
+            auto direction = gMap_507BA8.GetDirection(
+                field_B2_lvl_number,
+                field_B0_path_number,
+                field_A8_xpos,
+                field_AC_ypos);
+            SFX_Play_43AED0(3u, 50, direction);
+
+            if (((field_114_timer - gnFrameCount_507670) & ~7) >= 18 * 8)
```
Comment (paulsapps) [inline Source/AliveLibAO/TimedMine.cpp:313]: add a comment saying its rounding to multiples of 8 so we don't have to figure this out again?

Comment (d3nwah) [reply Source/AliveLibAO/TimedMine.cpp:314]: Added

**Pull request merged**